File: 品味Java的21種設計模式/dp_cpp/memento/example4/SubstractCommand.cpp
#include "SubstractCommand.h"

namespace cn {
namespace javass {
namespace dp {
namespace memento {
namespace example4 {

SubstractCommand::SubstractCommand(int opeNum) {
    this->opeNum = opeNum;
}

void SubstractCommand::execute() {
    this->operation->substract(opeNum);
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/memento/example4/SubstractCommand.java
package cn.javass.dp.memento.example4;

public class SubstractCommand extends AbstractCommand {

    private int opeNum;
    public SubstractCommand(int opeNum) {
        this.opeNum = opeNum;
    }
    public void execute() {
        this.operation.substract(opeNum);
    }

}
